kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -101,7 +105,11 @@ impl<'tcx> GotocCtx<'tcx> {
         loc: Location,
     ) -> Stmt {
         let property_name = property_class.as_str();
-        Stmt::assert(cond, property_name, message, loc)
+        if self.queries.check_coverage && property_class != PropertyClass::Coverage {
+            Stmt::assume(cond, loc)
```

Comment:
Right, but is this function used for instrumenting checks? I checked the code and found that it's mainly used for cover and reachability checks (which we should **not** use an assumption for). The only case it's used for checks is https://github.com/model-checking/kani/blob/319d859bf2fc32e6025b9844538615f612b9788e/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs#L193
but this should probably be fixed to use `codegen_assert_assume` like all other checks.

Reply:
The change was undone recently (i.e., we don't replace the `assert` anymore). I also think we should be using `codegen_assert_assume` in the intrinsic (otherwise there should be a comment explaining the exception), so I've fixed it.